src/js/player.js

```diff
@@ -1562,6 +1572,10 @@ class Player extends Component {
   techGet_(method) {
     if (this.tech_ && this.tech_.isReady_) {
 
+      if (method in middleware.allowedGetters) {
```

Comment:
we can't actually do this cleanup here because we need to return a value from this function and `this.ready()` doesn't guarantee a value, even if you're running it synchronously. However, I think I was able to simplify `techCall`.